pkg/configuration/configuration.go

```diff
@@ -0,0 +1,94 @@
+// Package configuration is in charge of the validation and extraction of all
+// the configuration details from a configuration file or environment variables.
+package configuration
+
+import (
+	"os"
+	"strings"
+
+	errs "github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+// prefixes
+const (
+	// MemberEnvPrefix will be used for member environment variable name prefixing.
+	MemberEnvPrefix = "MEMBER_OPERATOR"
+)
+
+// Configuration constants
+const (
+	// ToolchainConfigMapName specifies a name of a ConfigMap that keeps toolchain configuration
+	ToolchainConfigMapName = "toolchain-saas-config"
+
+	// IdentityProvider specifies an identity provider (IdP) for newly created users
+	IdentityProvider = "identity.provider"
+
+	// DefaultIdentityProvider the default value used for the identity provider (IdP) for newly created users
+	DefaultIdentityProvider = "rhd"
+)
+
+// Registry encapsulates the Viper configuration registry which stores the
+// configuration data in-memory.
+type Registry struct {
+	member *viper.Viper
+}
+
+// CreateEmptyRegistry creates an initial, empty registry.
+func CreateEmptyRegistry() *Registry {
+	c := Registry{
+		member: viper.New(),
+	}
+	c.member.SetEnvPrefix(MemberEnvPrefix)
+	c.member.AutomaticEnv()
+	c.member.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	c.setConfigDefaults()
+	return &c
+}
+
+// New creates a configuration reader object using a configurable configuration
+// file path. If the provided config file path is empty, a default configuration
+// will be created.
+func New(configFilePath string) (*Registry, error) {
+	c := CreateEmptyRegistry()
+	if configFilePath != "" {
```

Comment:
same here, I believe we can skip the support for an external config file. We'll just use config maps and env vars, AFAIK.

Reply:
This is also something taken from the host operator. Are we sure we want to remove it? I can make the same change to the host operator when we've confirmed.

Reply:
yes, we don't need to load the config from the files - you can drop it. 
Thanks and sorry for bringing the confusions :-)

Reply:
Removed as part of a0acc7e66f8b3bca14fd5cad1631b966e28b399d